lib/ui/screens/coffee_screen.dart

```diff
@@ -61,14 +63,14 @@ class _CoffeeScreenState extends State<CoffeeScreen> {
   List<ShotState> dataPoints = [];
   double baseTime;
 
-  List<charts.Series<ShotState, double>> _createData() {
+  List<charts.Series<ShotState, double>> _createData(FluTheme theme) {

```

Comment:
I really dislike introducing the theme as a state that is to be changed.
After all the theme is global an dsingleton in logic. Not sure if we are not overcomplicating it here.

Reply:
I think quite a few of these could probably be removed entirely depending on how we populate the default ThemeData. But I think passing in the `context` here is probably the end game for having utility build methods which will provide all the data / theme / etc.. we may need. Passing down is always better than globals.